src/t8_geometry/t8_geometry.h

```diff
@@ -56,138 +56,60 @@ typedef enum t8_geometry_type {
+ */
+typedef struct t8_geometry_handler t8_geometry_handler_c;
+
 /* The t8_geometry_c type must be know to cmesh.h, thus we
  * include it after the typedef. */
 #include <t8_cmesh.h>
 
-typedef struct t8_geometry_handler
-{
-  sc_array_t registered_geometries;
-  /**< Stores all geometries that are handled by this geometry_handler. */
-  t8_geometry_c *active_geometry;
-  /**< Points to the currently loaded geometry (the geometry that was used last and is likely to be used next). */
-  t8_gloidx_t active_tree;
-  /**< The global tree id of the last tree for which geometry was used. */
-  int is_committed;
-  /**< If true, no new geometries can be registered. */
-  t8_refcount_t rc;
-  /**< The reference count of the geometry handler. */
-} t8_geometry_handler_t;
-
 T8_EXTERN_C_BEGIN ();
 
-/** 
- * Initialize a geometry handler. This will allocate memory.
- * \param [in,out]  pgeom_handler On input a pointer to unallocated memory.
- *                                On output this will point to an initialized geometry handler.
- */
-void
-t8_geom_handler_init (t8_geometry_handler_t **pgeom_handler);
-
-/** 
- * Increase the reference counter of a geometry handler.
- * \param [in] geom_handler An initialized geometry handler.
- */
-void
-t8_geom_handler_ref (t8_geometry_handler_t *geom_handler);
-
-/**
- * Decrease the reference count of a geometry handler, if the refcount
- * reaches 0, the handler will get destroyed.
- * \param [in,out] pgeom_handler Pointer to an initialized geometry handler.
- *                               If the refcount reaches 0, will point to NULL afterwards.
- */
-void
-t8_geom_handler_unref (t8_geometry_handler_t **pgeom_handler);
-
-/**
- * Destroy a geometry handler and free its memory. 
- * This is only valid if its reference count is 1.
- * If you are unsure, call \ref t8_geom_handler_unref instead.
- * \param [in,out] pgeom_handler Pointer to an initialized geometry handler.
- *                               Will point to NULL afterwards.
- */
-void
-t8_geom_handler_destroy (t8_geometry_handler_t **pgeom_handler);
-
-/**
- * Add a geometry to the geometry handler.
- * \param [in,out] geom_handler An initialized but not committed geometry handler.
- * \param [in]     geometry     The geometry to add.
- */
-void
-t8_geom_handler_register_geometry (t8_geometry_handler_t *geom_handler, const t8_geometry_c *geometry);
-
 /**
- * Commit a geometry handler. This specifies that no geometries will
- * be added to it and makes it ready to be used.
- * \param [in,out] geom_handler An initialized but not committed geometry handler.
+ * Evaluates the geometry of a tree at a given reference point.
+ * \param [in] cmesh The cmesh
+ * \param [in] gtreeid The global id of the tree
+ * \param [in] ref_coords The reference coordinates at which to evaluate the geometry
+ * \param [in] num_coords The number of reference coordinates
+ * \param [out] out_coords The evaluated coordinates
```

Comment:
Not crucial, but i like it when the param names and the descriptions are indented properly 😁